external-crates/move/crates/move-analyzer/src/inlay_hints.rs

```diff
@@ -117,3 +121,62 @@ fn additional_hint_info(sp!(_, t): &N::Type, symbols: &Symbols) -> Option<InlayH
         struct_def_info,
     )))
 }
+
+#[cfg(test)]
+fn validate_type_hint(hints: &[InlayHint], path: &std::path::Path, line: u32, col: u32, ty: &str) {
+    let lsp_line = line - 1; // 0th based
+    if let Some(label_parts) = hints.iter().find_map(|h| {
+        if h.position.line == lsp_line && h.position.character == col {
+            if let InlayHintLabel::LabelParts(parts) = &h.label {
+                return Some(parts);
+            }
+        }
+        None
+    }) {
+        let found_ty = &label_parts[1].value;
+        assert!(
+            found_ty == ty,
+            "incorrect type of hint (found '{found_ty}' instead of expected '{ty}') at line {line} and col {col} in {path:?}"
+        );
+    } else {
+        panic!("hint not found at line {line} and col {col} in {path:?}");
+    }
+}
```

Comment:
When the nesting gets deep with `if let`s it sometimes helps to pull it out with a pattern like this, which I think reads better (note: untested, but I think this checks out--only mentioning because I've been running into this myself a lot recently)

```rust
let label_parts = hints.iter().find_map(|h| {
    if h.position.line == lsp_line && h.position.character == col {
        if let InlayHintLabel::LabelParts(parts) = &h.label {
            return Some(parts);
        }
    }
    None
}).unwrap_or_else(|| {
    panic!("hint not found at line {line} and col {col} in {path:?}");
});

let found_ty = &label_parts[1].value;
assert!(
    found_ty == ty,
    "incorrect type of hint (found '{found_ty}' instead of expected '{ty}') at line {line} and col {col} in {path:?}"
);
```

Reply:
Good point! I try to pay attention to this but this one clearly sneaked out. I think an even better pattern here (which I used) is `let-else` which allows one to avoid calling `unwrap_or_else`